### PR TITLE
Update docs about `show_progress_bar`

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

`show_progress_bar` is not experimental now, but docs is not updated.

## Description of the changes

Update the description.

### CODE REVIEW

**Comment by nabenabe0928** [comment]
@not522 Could you review this PR?

File: optuna/study/study.py [hunk 1]
```diff
@@ -439,10 +439,9 @@ def objective(trial):
                     :ref:`out-of-memory-gc-collect`
 
             show_progress_bar:
-                Flag to show progress bars or not. To disable progress bar, set this :obj:`False`.
-                Currently, progress bar is experimental feature and disabled
-                when ``n_trials`` is :obj:`None`, ``timeout`` is not :obj:`None`, and
-                ``n_jobs`` :math:`\\ne 1`.
+                Flag to show progress bars or not. To show progress bar, set this :obj:`True`.
+                Note that it is disabled when ``n_trials`` is :obj:`None`,
+                ``timeout`` is not :obj:`None`, and ``n_jobs`` :math:`\\ne 1`.
```
Comment (nabenabe0928) [inline optuna/study/study.py:444]: Isn't it `or`, meaning we cannot use the progress bar if any one of them is `True`? 
```suggestion
                ``timeout`` is not :obj:`None`, or ``n_jobs`` :math:`\\ne 1`.
```

**Review by nabenabe0928 — COMMENTED** [review]
I left a minor comment, but otherwise, LGTM!

File: optuna/study/study.py (hunk 1 above)
Comment (eukaryo) [reply optuna/study/study.py:444]: https://github.com/optuna/optuna/blob/master/optuna/study/_optimize.py#L52
According to the current implementation, `and`  is correct. Iff we allow unlimited trial, set a time-limit, and make multiple jobs, then `progress_bar` is disabled.

Comment (nabenabe0928) [reply optuna/study/study.py:444]: Thank you for the confirmation!

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Review by not522 — APPROVED** [review]
Thank you. LGTM.
It missed in #4684.

**Pull request merged**